web/src/engine/websites/Pururin.ts

```diff
@@ -0,0 +1,79 @@
+import { Tags } from '../Tags';
+import icon from './Pururin.webp';
+import { type Chapter, DecoratableMangaScraper, Page, Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import {FetchJSON, FetchRequest } from './../FetchProvider';
+
+type APIChapter = {
+    gallery: { total_pages: number, image_extension : string, title: string }
+}
+
+const CDN = 'https://cdn.pururin.to/assets/images/data';
+const apiPath = '/api/contribute/gallery/info';
+
+function MangaInfoExTractor(anchor: HTMLAnchorElement) {
+    const id = anchor.pathname;
+    const title = anchor.querySelector('img.card-img-top').getAttribute('alt').trim();
+    return { id, title };
+}
+
+@Common.MangasMultiPageCSS('/browse/title?page={page}', 'a.card.card-gallery', 1, 1, 0, MangaInfoExTractor)
+@Common.ChaptersUniqueFromManga()
+@Common.ImageDirect()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('pururin', `Pururin`, 'https://pururin.to', Tags.Language.English, Tags.Media.Manga, Tags.Source.Aggregator, Tags.Rating.Erotica);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /https?:\/\/pururin\.to/.test(url);
```

Comment:
Is it possible to use more strict regex? This may produce false positives and user may start asking why the pasted URL will fail.